rdt-app/src/main/java/io/ona/rdt_app/interactor/PatientRegisterFragmentInteractor.java

```diff
@@ -152,14 +146,16 @@ private EventClient saveEventClient(JSONObject jsonForm, String encounterType, S
     }
 
     /**
-     * Get Patient for RDT
+     * Get the patient for whom the RDT is to be conducted
      *
-     * @return Patient object for which the RDT capture is to be done.
+     * @param jsonForm The patient form JSON String
+     * @return the initialized Patient if proceeding to RDT capture otherwise return null patient
      * @throws JSONException
      */
-    public Patient getPatientForRDT() throws JSONException {
+    public Patient getPatientForRDT(String jsonForm) throws JSONException {
         Patient rdtPatient = null;
-        if (jsonFormObject != null && PATIENT_REGISTRATION.equals(jsonFormObject.optString(ENCOUNTER_TYPE))) {
```

Comment:
Could you revert this to the previous `jsonFormObject` global variable implementation?

The idea here is to avoid unnecessary string to json object conversions, which can be expensive.

If this is done for testing, you can always use PowerMock's `Whitebox` to set private variables.